doc/CMakeLists.txt

```diff
@@ -118,61 +120,103 @@ endif()
+        COMMENT "Building ${w_target}"
+        VERBATIM)
+    list(APPEND WEB_PAGE_OUTPUTS ${w_target})
 endforeach()
 add_custom_target(web-site ALL DEPENDS ${WEB_PAGE_OUTPUTS})
 
 # Reference Manual Components
 set(REF_PARTS ${CMAKE_CURRENT_BINARY_DIR}/etc/ref-parts.so)
 add_custom_command(OUTPUT ${REF_PARTS}
-  COMMAND ${REF_PTX} ${MAN_PAGES} > ${REF_PARTS}
-  DEPENDS ${MAN_PAGES} ${CMAKE_CURRENT_SOURCE_DIR}/script/ref-ptx.sh
-  COMMENT "Building ${REF_PARTS}"
-  VERBATIM
-  )
+    COMMAND
+        sh ${CMAKE_CURRENT_SOURCE_DIR}/script/ref-ptx.sh
+        ${MAN_PAGES}
+        > ${REF_PARTS}
+    DEPENDS
+        ${MAN_PAGES}
+        ${CMAKE_CURRENT_SOURCE_DIR}/script/ref-ptx.sh
+    COMMENT "Building ${REF_PARTS}"
+    VERBATIM)
 
 set(NEW_CHANGES ${CMAKE_CURRENT_BINARY_DIR}/etc/new.so)
 add_custom_command(OUTPUT ${NEW_CHANGES}
-  COMMAND ${NEW_CH} -r ${NEW_CHANGE_NOTES} > ${NEW_CHANGES}
-  DEPENDS ${NEW_CHANGE_NOTES} ${CMAKE_CURRENT_SOURCE_DIR}/script/ref-ptx.sh
-  COMMENT "Building ${NEW_CHANGES}"
-  VERBATIM
-  )
+    COMMAND
+        sh ${CMAKE_CURRENT_SOURCE_DIR}/script/new.sh
+        -r ${NEW_CHANGE_NOTES}
+        > ${NEW_CHANGES}
+    DEPENDS
+        ${NEW_CHANGE_NOTES}
+        ${CMAKE_CURRENT_SOURCE_DIR}/script/new.sh
+    COMMENT "Building ${NEW_CHANGES}"
+    VERBATIM)
 
 # Standalone Documents
 
 # Transform man page(s) (ie troff format files) into a PDF
-FUNCTION(ADD_DOC TARGET SOURCE DEPS)
-  add_custom_command(OUTPUT ${TARGET}
-    COMMAND ${GROFF} -Tps -s -I${CMAKE_CURRENT_SOURCE_DIR_UNIX_FORMAT} -I${CMAKE_CURRENT_BINARY_DIR_UNIX_FORMAT} -t -man ${SOURCE} > ${TARGET}.ps.tmp
-    COMMAND ${PAGE_SELECT} ${CMAKE_CURRENT_SOURCE_DIR}/etc/page-list.awk ${TARGET}.ps.tmp ${TARGET}.ps
-    COMMAND ${PS2PDFEXE} ${TARGET}.ps ${TARGET}
-    COMMAND rm ${TARGET}.ps ${TARGET}.ps.tmp
-    DEPENDS ${SOURCE} ${DEPS}
-    COMMENT "Building ${TARGET}"
-    VERBATIM)
-  list(APPEND DOC_TARGETS ${TARGET})
-  set(DOC_TARGETS ${DOC_TARGETS} PARENT_SCOPE)
-ENDFUNCTION()
-
-ADD_DOC(${CMAKE_CURRENT_BINARY_DIR}/BUILDING.pdf ${CMAKE_CURRENT_SOURCE_DIR}/etc/BUILDING.man "")
-ADD_DOC(${CMAKE_CURRENT_BINARY_DIR}/README.pdf ${CMAKE_CURRENT_SOURCE_DIR}/etc/README.man "${NEW_CHANGES}")
-ADD_DOC(${CMAKE_CURRENT_BINARY_DIR}/change_log.pdf ${CMAKE_CURRENT_SOURCE_DIR}/etc/change_log.man "${NEW_CHANGES}")
-ADD_DOC(${CMAKE_CURRENT_BINARY_DIR}/reference-${srecord_VERSION_MAJOR}.${srecord_VERSION_MINOR}.pdf
-        ${CMAKE_CURRENT_SOURCE_DIR}/etc/reference.man "${REF_PARTS};${NEW_CHANGES}")
+function(add_doc TARGET SOURCE DEPS)
+    add_custom_command(OUTPUT ${TARGET}
+        COMMAND
+            sh ${CMAKE_CURRENT_SOURCE_DIR}/script/groff.sh
+            -T ps                  # Output device
+            -man                   # Macro Package man
+            # Order of preprocessors may be important
+            -t                     # Preprocess with tbl
+            -s                     # Preprocess with soelim, with search paths:
```

Comment:
You write the order may be important, but swap it here in this commit which is mostly about style changes.

Reply:
I don't know groff, and found no hint about this behavior in the options description in the man page.

But checking the output for being identical I recognized that the order of `-P-I${image_stem_name}` had an influence on the filename of the generated png.

Reply:
I did now normalize the order of all groff calls. They order in general seems to be irrelevant, just that one is not.
I left the comment in the code, so the people are aware to be sure to check the output files if those things are changed.

Reply:
Yep, the ordering is important except when it isn't :-(. Adding the image stem option was fiddly. Needed to give unique website images but also to support concurrent doc builds (there was a race condition around generating an image for one doc entry being overwritten by another running in parallel). The other main gotchas I encountered were around the args being passed to the preprocesors like soelim and tbl.